spec/features/user_searches_spec.rb

```diff
@@ -0,0 +1,141 @@
+      end
+    end
+
+    it "dosen't see results that are not matched" do
+      setup_test_data
+      ghost = create(:node, label: "Node ghost")
+      term = "search"
+
+      page.find(".form-search #q").set(term)
+      click_on "search-btn"
+
+      within "#tbl-search" do
+        expect(page).to_not have_content ghost.label
+      end
+    end
+
+    it "clicking on node tab sees only matched nodes" do
+      setup_test_data
+      term = "search"
+      page.find(".form-search #q").set(term)
+      click_on "search-btn"
+
+      page.find(".search-nav #nodes").click
+
+      within "#tbl-search" do
+        expect(page).to have_css '.node-result'
+
+        expect(page).to_not have_css '.issue-result'
+        expect(page).to_not have_css '.note-result'
+        expect(page).to_not have_css '.evidence-result'
+      end
+    end
+
+    it "clicking on note tab sees only matched notes" do
+      setup_test_data
+      term = "search"
+      page.find(".form-search #q").set(term)
+      click_on "search-btn"
+
+      page.find(".search-nav #notes").click
+
+      within "#tbl-search" do
+        expect(page).to have_css '.note-result'
+
+        expect(page).to_not have_css '.node-result'
+        expect(page).to_not have_css '.issue-result'
+        expect(page).to_not have_css '.evidence-result'
+      end
+    end
+
+    it "clicking on issues tab sees only matched issues" do
+      setup_test_data
+      term = "search"
+      page.find(".form-search #q").set(term)
+      click_on "search-btn"
+
+      page.find(".search-nav #issues").click
+
+      within "#tbl-search" do
+        expect(page).to have_css '.issue-result'
+
+        expect(page).to_not have_css '.note-result'
+        expect(page).to_not have_css '.node-result'
+        expect(page).to_not have_css '.evidence-result'
+      end
+    end
+
+    it "clicking on evidences tab sees only matched evidences" do
+      setup_test_data
+      term = "search"
+      page.find(".form-search #q").set(term)
+      click_on "search-btn"
+
+      page.find(".search-nav #evidences").click
+
+      within "#tbl-search" do
+        expect(page).to have_css '.evidence-result'
+
+        expect(page).to_not have_css '.issue-result'
+        expect(page).to_not have_css '.note-result'
+        expect(page).to_not have_css '.node-result'
+      end
+    end
+
+    it "sees message warning when no search criteria entered" do
+      click_on "search-btn"
+
+      expect(page).to have_content 'Please enter search criteria'
+    end
+
+    it "sees message warning when no matches find" do
+      page.find(".form-search #q").set("no matches")
+
+      click_on "search-btn"
+
+      expect(page).to have_content 'No matches found!'
+    end
+  end
+end
+
+def setup_test_data
+  create(:issue, text: "Issue search")
+  node = create(:node, label: "Node search")
+  create(:note, text: "Note search", node: node)
+  create(:evidence, content: "Evidence search")
+end
```

Comment:
This is defined in global scope and will potentially pollute/conflict with other specs. Better to define it within the file's outermost `describe` block.